krabs/krabs/trace.hpp

```diff
@@ -233,6 +234,8 @@ namespace krabs {
 
         EVENT_TRACE_PROPERTIES properties_;
 
+        schema_locator schema_locator_;
```

Comment:
Keep a trace_context instead, Make it mutable so we can keep everything const on the API.
```suggestion
        mutable trace_context context_;
```